packages/devtools_app/lib/src/screens/memory/framework/memory_controller.dart

```diff
@@ -91,22 +96,21 @@ class MemoryController extends DisposableController
       case DevToolsMode.offlineData:
         assert(connectedDiff == null && connectedProfile == null);
         await maybeLoadOfflineData(
-          PerformanceScreen.id,
+          ScreenMetaData.memory.id,
           createData: (json) => OfflineMemoryData.parse(json),
-          shouldLoad: (data) => !data.isEmpty,
+          shouldLoad: (data) => true,
         );
-        // If shouldLoad returns false, previous line is noop, so data should be initialized.
-        if (!isInitialized.value) _initializeData();
+        assert(_initialized.isCompleted);
     }
-    assert(isInitialized.value);
+    assert(_initialized.isCompleted);
   }
 
   void _initializeData({
     OfflineMemoryData? offlineData,
     @visibleForTesting DiffPaneController? diffPaneController,
     @visibleForTesting ProfilePaneController? profilePaneController,
   }) {
-    assert(!isInitialized.value);
+    assert(!_initialized.isCompleted);
```

Comment:
we don't need this assert. `_initialized.complete();` will fail if the completer has already been completed

Reply:
`_initialized.complete()` happens 20 lines later. 
This assert helps with readability as it shows precondition of the method.